widget/locale.go

```diff
@@ -0,0 +1,90 @@
+package widget
+
+import (
+	"strings"
+
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/lang"
+)
+
+const (
+	localeDateFormatKey = "dateformat"
+	localeWeekStartKey  = "startofweek"
+)
+
+var localeSettings = map[string]map[string]string{
+	"": {
+		localeDateFormatKey: "02/01/2006",
+		localeWeekStartKey:  "Monday",
+	},
+	"BR": {
+		localeWeekStartKey: "Sunday",
+	},
+	"BZ": {
+		localeWeekStartKey: "Sunday",
+	},
+	"CA": {
+		localeWeekStartKey: "Sunday",
+	},
+	"CO": {
+		localeWeekStartKey: "Sunday",
+	},
+	"DO": {
+		localeWeekStartKey: "Sunday",
+	},
+	"GT": {
+		localeWeekStartKey: "Sunday",
+	},
+	"JP": {
+		localeWeekStartKey: "Sunday",
+	},
+	"MX": {
+		localeWeekStartKey: "Sunday",
+	},
+	"NI": {
+		localeWeekStartKey: "Sunday",
+	},
+	"PE": {
+		localeWeekStartKey: "Sunday",
+	},
+	"PA": {
+		localeWeekStartKey: "Sunday",
+	},
+	"PY": {
+		localeWeekStartKey: "Sunday",
+	},
+	"US": {
+		localeDateFormatKey: "01/02/2006",
+		localeWeekStartKey:  "Sunday",
+	},
+	"VE": {
+		localeWeekStartKey: "Sunday",
+	},
+	"ZA": {
+		localeWeekStartKey: "Sunday",
+	},
+}
+
+func getLocaleDateFormat() string {
+	return lookupLocaleSetting(localeDateFormatKey, lang.SystemLocale())
+}
+
+func getLocaleWeekStart() string {
+	return lookupLocaleSetting(localeWeekStartKey, lang.SystemLocale())
+}
+
+func lookupLocaleSetting(key string, l fyne.Locale) string {
+	region := ""
+	lang := l.LanguageString()
+	if pos := strings.Index(lang, "-"); pos != -1 {
+		region = strings.Split(lang, "-")[1]
+	}
```

Comment:
I think this might be cleaner to do using `strings.Cut`.

Reply:
Doesn't that require the second instance of the separator? It is optional in this instance.

Reply:
I figured it would be possible to do something like this to just use one strings API instead of two to achieve the same result. However, I do see now that the following code assumes that the region is the end of the file so it might be moot.

```
_, trailing, found := strings.Cut(lang, "-")
if found {
    region = trailing
}
```